### PR TITLE
PXP-7074 fix(google): when linking an external bucket, search for existing buc…

### PR DESCRIPTION
**Pull request by Avantol13**
…ket before creating another

Description about what this pull request does.

Please make sure to follow the [DEV guidelines](https://gen3.org/resources/developer/dev-introduction/) before asking for review.

### Bug Fixes
- Fix issue where duplicate bucket entries could get created in the database if you accidentally called `fence-create link-external-bucket` more than once

### CODE REVIEW

File: fence/scripting/fence_create.py [hunk 1]
```diff
@@ -1318,6 +1318,38 @@ def link_external_bucket(db, name):
...
+                    f"Existing bucket {name} has more than 1 associated "
+                    "Google Bucket Access Group with privilege of 'read'. "
+                    "This is not expected and we cannot continue linking."
+                )
+            elif len(access_group) == 0:
+                raise Exception(
+                    f"Existing bucket {name} has no associated "
+                    "Google Bucket Access Group with privilege of 'read'. "
+                    "This is not expected and we cannot continue linking."
+                )
```
Comment (paulineribeyre) [inline fence/scripting/fence_create.py:1341]: is there no edge case where you'd first create a bucket and later link it?

File: fence/scripting/fence_create.py [hunk 2]
```diff
@@ -1318,6 +1318,38 @@ def link_external_bucket(db, name):
...
+
+            access_group = access_group[0]
+
+            email = getattr(access_group, "email", "N/A")
```
Comment (paulineribeyre) [inline fence/scripting/fence_create.py:1345]: why not just `access_group.email`? is a gbag without an email valid? [we define it as non nullable](https://github.com/uc-cdis/fence/blob/2020.11/fence/models.py#L420)

File: fence/scripting/fence_create.py (hunk 1 above)
Comment (Avantol13) [reply fence/scripting/fence_create.py:1341]: the bucket and group should get created at the same time. So having a google bucket without a group is not something that should be able to happen under normal circumstances https://github.com/uc-cdis/fence/blob/2a3e7346579eb8cf5e628d116b6c4402b4e33c93/fence/scripting/fence_create.py#L1086-L1093

Comment (Avantol13) [reply fence/scripting/fence_create.py:1341]: so it's possible for a bucket to exist without being linked to a project, but there should always be a bucket+group combination

File: fence/scripting/fence_create.py (hunk 2 above)
Comment (Avantol13) [reply fence/scripting/fence_create.py:1345]: oh, this was before I checked the length and the result of query could be `None`. I'll change it

**Review by paulineribeyre — APPROVED** [review]

**Pull request merged**